python/lsst/ts/mtaos/mtaos.py

```diff
@@ -1006,6 +1018,24 @@ async def run_closed_loop(self):
                         self.log.info(f"Visit {visit_id} already processed, skipping.")
                         continue
 
+                    filter_label, elevation = await self.model.get_image_info(
+                        visit_id,
+                        self.camera_name,
+                    )
+
+                    rotation_angle = float(
+                        np.mean(np.array(self.image_rotator[image_in_oods.obsid]))
+                    )
+
```

Comment:
Keep in mind that while we are moving on here, `self.current_rotator_position` and `self.current_elevation_position` are being updated in the background. I suggest you make a local variable to store a static value:

```
current_rotator_position = self.current_rotator_position
current_elevation_position = self.current_elevation_position
```

and use the local variable from here on out.